src/build.rs

```diff
@@ -112,95 +154,73 @@ pub struct AggregatedConfig {
     pub prepend_js: Vec< (PathBuf, String) >
 }
 
-impl BuildArgs {
-    pub fn new( matches: &clap::ArgMatches ) -> Result< Self, Error > {
-        let features = if let Some( features ) = matches.value_of( "features" ) {
-            features.split_whitespace().map( |feature| feature.to_owned() ).collect()
-        } else {
-            Vec::new()
-        };
-
-        let no_default_features = matches.is_present( "no-default-features" );
-        let enable_all_features = matches.is_present( "all-features" );
-
-        let build_type = if matches.is_present( "release" ) {
-            BuildType::Release
+impl From<super::Build> for BuildArgs {
+    fn from(b: super::Build) -> Self {
+        let backend = if let Some(be) = b.target {
+            be
         } else {
-            BuildType::Debug
-        };
-
-        let use_system_emscripten = matches.is_present( "use-system-emscripten" );
-        let is_verbose = matches.is_present( "verbose" );
-        let message_format = if let Some( name ) = matches.value_of( "message-format" ) {
-            match name {
-                "human" => MessageFormat::Human,
-                "json" => MessageFormat::Json,
-                _ => unreachable!()
+            if b.target_webasm_emscripten {
+                eprintln!("The `--target-webasm-emscripten` flag is DEPRECATED. \
+                           Please use the `--target` option with the full triple \
+                           (`\"{}\"`)", WASM_EMCC);
```

Comment:
Won't those strings have a bunch of spaces in the middle?

Reply:
(i've moved this code into `main.rs`)

the backslashes continue the lines [per the reference](https://doc.rust-lang.org/reference/tokens.html#string-literals)